src/main/scala/midas/models/dram/MidasMemModel.scala

```diff
@@ -375,6 +377,19 @@ class MidasMemModel(cfg: BaseConfig)(implicit p: Parameters) extends MemModel {
     attachIO(iWriteLatencyHist, "ingressWriteLatencyHist_")
   }
 
+  if (cfg.params.addrRangeCounters > 0) {
+    val n = cfg.params.addrRangeCounters
+    val readRanges = AddressRangeCounter(n, model.io.tNasti.ar, targetFire)
+    val writeRanges = AddressRangeCounter(n, model.io.tNasti.aw, targetFire)
+    val addrBits = model.io.tNasti.nastiXAddrBits
+    val numRanges = n.U(addrBits.W)
+
+    attachIO(readRanges, "readRanges_")
+    attachIO(writeRanges, "writeRanges_")
+    attach(numRanges(31, 0), "numRangesL", ReadOnly)
```

Comment:
I feel like there should be a considerably smaller limit on the number of bins -- you probably only need one register.

Reply:
Okay, I'll just make it 32 bits.

Reply:
Maybe also add a requirement on the max size. Something you think is reasonable.